packages/fxa-settings/src/lib/hooks.tsx

```diff
@@ -55,3 +57,13 @@ export function useChangeFocusEffect() {
 
   return elToFocus;
 }
+
+/*
+ * Temporary helper to check that emailFirstRoutes feature flag is on
+ * and (if not 100% rolled out) that the user is in the React experiment.
+ */
+export function useCheckReactEmailFirst() {
+  const config = useConfig();
+  // TODO with FXA-11221 (100% prod rollout), remove isInReactExperiment() check
+  return config.showReactApp.emailFirstRoutes === true && isInReactExperiment();
```

Comment:
Here are the options I debated for this:
- (what I went with) When we roll out React email first to 100%, which requires a 1-line code change, we make it a 2-line change instead and remove an `isInReactExperiment()` check
- More work to access `fullProdRollout` from content-server `react-app/index.js`
- Checking query param for `prefillEmail` / all other references, and make follow up to check location state later
- Keep the hard navigate and hard reload the React bundle until we clean up later after stabilization
- Take users to React email-first if feature flag is on, even if they aren't in the experiment